src/host/utf8handling.c

```diff
@@ -0,0 +1,33 @@
+/**
+ * \file   utf8handking.c
+ * \brief  Handles conversions between Unicode (UTF-8) and system native encoding (wide chars on Windows)
+ * \author Copyright (c) 2017 Jérôme Leclercq and the Premake project
+ */
+
+#include "premake.h"
+#include "stdlib.h"
+
+#ifdef PLATFORM_WINDOWS
+const char* utf8_fromwide(lua_State* L, const wchar_t* wstr)
```

Comment:
I would prefer a ```size_t utf8_fromwide(char* destination, size_t destSize, const wchar_t* wstr);```

which would not need to allocate memory, would allow for the call to WideCharToMultiByte to simply be called once. And wouldn't need the LuaState... pushing the string to the state would be up to the caller.